karabo_data/geometry2.py

```diff
@@ -58,16 +63,18 @@ def snap(self):
         fs_vec = np.around(self.fs_vec[:2]).astype(np.int32)
         assert {tuple(np.abs(ss_vec)), tuple(np.abs(fs_vec))} == {(0, 1), (1, 0)}
         # Convert xy coordinates to yx indexes
-        return GridGeometryFragment(corner_pos[::-1], ss_vec[::-1], fs_vec[::-1])
+        return GridGeometryFragment(corner_pos[::-1], ss_vec[::-1], fs_vec[::-1],
+                                    self.ss_pixels, self.fs_pixels)
 
-class GridGeometryFragment:
-    ss_pixels = 64
-    fs_pixels = 128
 
+class GridGeometryFragment:
```

Comment:
This class is important for snap/ideal geometry. It must have a docstring especially explaining flip without transpose or transpose then flip part.